dev/Scroller/TestUI/ScrollerDynamicPage.xaml.cs

```diff
@@ -2008,7 +2008,94 @@ private void BtnAddRepeatedSnapPoint_Click(object sender, RoutedEventArgs e)
             }
         }
 
-        private void BtnClearSnapPoints_Click(object sender, RoutedEventArgs e)
+        private void BtnChangeView_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                double? horizontalOffset = string.IsNullOrWhiteSpace(txtCOAHO.Text) ? (double?)null : Convert.ToDouble(txtCOAHO.Text);
+                double? verticalOffset = string.IsNullOrWhiteSpace(txtCOAVO.Text) ? (double?)null : Convert.ToDouble(txtCOAVO.Text);
+                float? zoomFactor = string.IsNullOrWhiteSpace(txtCZFAZF.Text) ? (float?)null : Convert.ToSingle(txtCZFAZF.Text);
+                bool disableAnimation = true;
+
+                if (horizontalOffset == null && verticalOffset == null && zoomFactor == null)
+                {
+                    return;
+                }
+
+                ComboBox cmbAnimationMode = (zoomFactor == null || zoomFactor == scroller.ZoomFactor) ? 
+                    cmbScrollAnimationMode : cmbZoomAnimationMode;
+
+                switch (cmbAnimationMode.SelectedIndex)
+                {
+                    case 0: // Disabled
+                        break;
+                    case 1: // Enabled
+                        disableAnimation = false;
+                        break;
+                    default: // Auto
+                        disableAnimation = !new UISettings().AnimationsEnabled;
+                        break;
+                }
+
+                ChangeView(scroller, horizontalOffset, verticalOffset, zoomFactor, disableAnimation);
+            }
+            catch (Exception ex)
+            {
+                txtExceptionReport.Text = ex.ToString();
+                lstScrollerEvents.Items.Add(ex.ToString());
+            }
+        }
+
+        private bool ChangeView(Scroller scroller, double? horizontalOffset, double? verticalOffset, float? zoomFactor, bool disableAnimation)
+        {
+            AppendAsyncEventMessage($"ChangeView(horizontalOffset:{horizontalOffset}, verticalOffset:{verticalOffset}, zoomFactor:{zoomFactor}, disableAnimation:{disableAnimation}) from horizontalOffset:{scroller.HorizontalOffset}, verticalOffset:{scroller.VerticalOffset}, zoomFactor:{scroller.ZoomFactor}");
+
+            double targetHorizontalOffset = horizontalOffset == null ? scroller.HorizontalOffset : (double)horizontalOffset;
+            double targetVerticalOffset = verticalOffset == null ? scroller.VerticalOffset : (double)verticalOffset;
+            float targetZoomFactor = zoomFactor == null ? scroller.ZoomFactor : (float)Math.Max(Math.Min((double)zoomFactor, scroller.MaxZoomFactor), scroller.MinZoomFactor);
+            float deltaZoomFactor = targetZoomFactor - scroller.ZoomFactor;
+
+            if (disableAnimation)
```

Comment:
Is the clamping required only when animation is disabled ? Also, would scroller not do this clamping for you ?

Reply:
Added a comment.   The InteractionTracker would ultimately do the clamping, however the computation of the centerPoint would be incorrect without that clamping.